Search for the same keyword over multiple locations.

ryanbillings opened the pull request:
This adds support for a single *keyword* string with an array of *geo* locations.

I ran into error `SyntaxError: Unexpected token C in JSON at position 0` when running a query such as:

```javascript
googleTrends.interestOverTime({
  keyword: 'javascript',
  startTime: new Date('2018-01-01'),
  geo: ['US-MA', 'US-VA']
});
```

My solution to this was to search for the same `keyword` for all regions in the case when keyword is a string while `geo` is an array.

@pat310 Let me know if you think this is a good change, thanks!

pat310 commented on src/utilities.js:
```diff
@@ -120,9 +120,16 @@ export function parseResults(results) {
  */
 export function formatComparisonItems(obj) {
+  const isMultiRegion = obj.geo && Array.isArray(obj.geo);
+  const isMultiKeyword = () => Array.isArray(obj.keyword);
```
Can we make this just a constant rather than a function since it doesn't actually take any parameters?
`const isMultiKeyword = Array.isArray(obj.keyword)`

ryanbillings replied:
Good call! Addressed in https://github.com/pat310/google-trends-api/pull/116/commits/d9090f4e336226b5260d1cd12ea092af9c862bf8

pat310 commented:
Looks good, just one comment! @ryanbillings

pat310 commented:
@ryanbillings How will this work if `keyword` is an array and `geo` is an array? Will it run every possible combination of `keyword` and `geo`?

ryanbillings commented:
Thanks for the PR review @pat310 !

The current implementation will run through both the `geo` and `keyword` arrays and run a query for each index in the array. If the arrays are mismatching lengths, we get an error (the one I posted above). To address this, I added in a validation in https://github.com/pat310/google-trends-api/pull/116/commits/30662b83578a92d3d923a1c720f6e99f93e60cd2 to make sure the arrays are the same size, along with some README updates to clarify this:

```markdown
Passing `geo: ['US-CA, US-VA'], keyword: ['wine', 'peanuts']` will search for wine in California and peanuts in Virginia.
```

Also moved all of the possible cases for creating errors into a `validateObj` function which gets called at the beginning of `constructObj` (needed to satisfy a codeclimate failure 🙂 )
